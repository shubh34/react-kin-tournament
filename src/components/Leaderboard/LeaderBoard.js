
import React from 'react';
import PropTypes from 'prop-types';
import Table from '../../sharedComponent/Table';

const LeaderBoard = ({ leaders, userId }) => (
	<Table>

		<Table.Body>
			<Table.Row>
				<Table.Header>Pos</Table.Header>
				<Table.Header>Score</Table.Header>
				<Table.Header>Prize</Table.Header>
			</Table.Row>
			{leaders.map(leader => (
				<Table.Row key={leader.playerId} isActive={leader.playerId === userId}>
					<Table.Column>
						{`${leader.position}.`}
					</Table.Column>
					<Table.Column>
						{leader.score}
					</Table.Column>
					<Table.Column>
						<strong>
							{leader.prize}
						</strong>

					</Table.Column>
				</Table.Row>
			))}
		</Table.Body>
	</Table>
);


LeaderBoard.propTypes = {
	leaders: PropTypes.arrayOf(PropTypes.object).isRequired,
	userId: PropTypes.number.isRequired,
};

export default LeaderBoard;
