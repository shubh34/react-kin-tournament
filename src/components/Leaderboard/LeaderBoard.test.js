import React from 'react';
import { shallow } from 'enzyme';

import LeaderBoard from './LeaderBoard';
import { defaultLeaders } from '../../../tests/leaders';

const setUp = (setUpProps = {}, container = shallow) => {
	const props = {
		leaders: defaultLeaders.slice(0, 4),
		userId: 18,
		...setUpProps,
	};
	const wrapper = container(<LeaderBoard {...props} />);
	return { wrapper, props };
};

describe('LeaderBoard', () => {
	it('should render leaders of tournament with user as active row', () => {
		const { wrapper } = setUp();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render leaders of tournament with no active user', () => {
		const { wrapper } = setUp({ userId: 110 });
		expect(wrapper).toMatchSnapshot();
	});
});
