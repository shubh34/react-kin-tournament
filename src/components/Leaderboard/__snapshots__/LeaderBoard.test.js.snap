// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard should render leaders of tournament with no active user 1`] = `
<Table>
  <TableBody>
    <TableRow>
      <TableHeader
        key="Pos"
      >
        Pos
      </TableHeader>
      <TableHeader
        key="Score"
      >
        Score
      </TableHeader>
      <TableHeader
        key="Prize"
      >
        Prize
      </TableHeader>
    </TableRow>
    <TableRow
      isActive={false}
      key="1"
    >
      <TableColumn>
        1.
      </TableColumn>
      <TableColumn>
        90
      </TableColumn>
      <TableColumn>
        <strong />
      </TableColumn>
    </TableRow>
    <TableRow
      isActive={false}
      key="2"
    >
      <TableColumn>
        2.
      </TableColumn>
      <TableColumn>
        86
      </TableColumn>
      <TableColumn>
        <strong />
      </TableColumn>
    </TableRow>
    <TableRow
      isActive={false}
      key="3"
    >
      <TableColumn>
        3.
      </TableColumn>
      <TableColumn>
        76
      </TableColumn>
      <TableColumn>
        <strong />
      </TableColumn>
    </TableRow>
    <TableRow
      isActive={false}
      key="4"
    >
      <TableColumn>
        4.
      </TableColumn>
      <TableColumn>
        75
      </TableColumn>
      <TableColumn>
        <strong />
      </TableColumn>
    </TableRow>
  </TableBody>
</Table>
`;

exports[`Leaderboard should render leaders of tournament with user as active row 1`] = `
<Table>
  <TableBody>
    <TableRow>
      <TableHeader
        key="Pos"
      >
        Pos
      </TableHeader>
      <TableHeader
        key="Score"
      >
        Score
      </TableHeader>
      <TableHeader
        key="Prize"
      >
        Prize
      </TableHeader>
    </TableRow>
    <TableRow
      isActive={false}
      key="1"
    >
      <TableColumn>
        1.
      </TableColumn>
      <TableColumn>
        90
      </TableColumn>
      <TableColumn>
        <strong />
      </TableColumn>
    </TableRow>
    <TableRow
      isActive={true}
      key="2"
    >
      <TableColumn>
        2.
      </TableColumn>
      <TableColumn>
        86
      </TableColumn>
      <TableColumn>
        <strong />
      </TableColumn>
    </TableRow>
    <TableRow
      isActive={false}
      key="3"
    >
      <TableColumn>
        3.
      </TableColumn>
      <TableColumn>
        76
      </TableColumn>
      <TableColumn>
        <strong />
      </TableColumn>
    </TableRow>
    <TableRow
      isActive={false}
      key="4"
    >
      <TableColumn>
        4.
      </TableColumn>
      <TableColumn>
        75
      </TableColumn>
      <TableColumn>
        <strong />
      </TableColumn>
    </TableRow>
  </TableBody>
</Table>
`;
